include/boost/geometry/formulas/spherical.hpp

```diff
@@ -278,6 +288,546 @@ inline result_direct<CT> spherical_direct(CT const& lon1,
+      -> 3 calls to asin
+      -> 6 multiplications
+      Also we gain the operations used to compute XTD:
+      -> 2 calls to sin
+      -> 1 call to asin
+      -> 1 call to abs
+      -> 2 multiplications
+      -> 1 division
+      So the total gains are:
+      -> 9 calls to sqrt/sin/asin
+      -> 1 call to abs
+      -> 8 multiplications
+      -> 1 division
+
+      Loses:
+      To compute a distance compatible with comparable::haversine<>
+      we need to perform a few more operations, namely:
+      -> 1 call to sin
+      -> 1 call to sqrt
+      -> 2 multiplications
+      -> 1 division
+      -> 1 addition
+      -> 2 subtractions
+
+      So roughly speaking the net gain is:
+      -> 8 fewer function calls and 3 fewer arithmetic operations
+
+      If we were to implement cross_track directly from the
+      comparable version (much like what haversine<> does using
+      comparable::haversine<>) we need additionally
+      -> 2 function calls (asin/sqrt)
+      -> 2 multiplications
+
+      So it pays off to re-implement cross_track<> to use
+      comparable::cross_track<>; in this case the net gain would be:
+      -> 6 function calls
+      -> 1 arithmetic operation
+
+   Summary/Conclusion
+   ------------------
+   Following the mathematical and complexity analysis above, the
+   comparable cross track strategy (as implemented below) satisfies
+   all the goal mentioned in the beginning:
+   * It is more efficient than its non-comparable counter-part.
+   * Comparable distances using this new strategy can also be compared
+     with comparable distances computed with the comparable haversine
+     strategy.
+   * It turns out to be more efficient to compute the actual cross
+     track distance XTD by first computing CXTD, and then computing
+     XTD by means of the formula:
+                XTD = 2 * R * asin( sqrt(CXTD) )
+*/
+
+
+/* Closest point to segment using cartesian geometry
+ *
+        typedef model::point<CT, 3, cs::cartesian> point3d_t;
+
+        point3d_t vp1 = formula::sph_to_cart3d<point3d_t>(sp1);
+        point3d_t vp2 = formula::sph_to_cart3d<point3d_t>(sp2);
+        point3d_t vp = formula::sph_to_cart3d<point3d_t>(p);
+
+        point3d_t G = cross_product(vp1, vp2);
+        point3d_t F = cross_product(G, vp);
+        //consider cases for G e.g.
+        set<0>(G, get<0>(G)*(-1));
+        set<1>(G, get<1>(G)*(-1));
+        set<2>(G, get<2>(G)*(-1));
+        point3d_t X = cross_product(G, F);
+        geometry::detail::vec_normalize(X);
+        Point result_p = formula::cart3d_to_sph<Point>(X);
+
+        closest_point_result.lon1 = get_as_radian<0>(p);
+        closest_point_result.lat1 = get_as_radian<1>(p);
+        closest_point_result.lon2 = get_as_radian<0>(result_p);
+        closest_point_result.lat2 = get_as_radian<1>(result_p);
+*/
+template <
+    typename CalculationType,
+    bool EnableClosestPoint = false
+>
+class comparable_spherical_point_segment_distance
```

Comment:
Here the name is ok.

Though this looks like a strategy or an algorithm. It takes the `CalculationType`, it takes `comparable_distance_strategy`, it defines `result_type`.

On the other hand it looks like at least parts of it could be a formula. Would it be possible to extract some kind of "formula core" from it? So a part only taking and returning coordinates, distances, etc. As other existing formulas do.

Reply:
I would like to leave it into strategies for now (removed it from formulas) and leave splitting of the "formula" parts for a new PR.